components/SocialSharing/SocialSharing.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from 'react';
+import { useCopyToClipboard } from 'usehooks-ts';
+import { Container } from '../Container';
+import { Heading } from '../Typography';
+import { FlexBox } from '../FlexBox';
+import { TwitterX } from './TwitterX';
+import { FacebookF } from './FacebookF';
+import { LinkedinIn } from './LinkedinIn';
+import { EmailIcon } from './EmailIcon';
+import { LinkIcon } from './LinkIcon';
+import { WidthBox } from '../WidthBox';
+import { SocialButton } from './SocialButton';
+import { HeroIcon } from '../HeroIcon';
+import { config } from '@/utilities/config';
+import * as styles from './SocialSharing.styles';
+
+type SocialSharingProps = React.HTMLAttributes<HTMLDivElement> & {
+  slug: string;
+  title?: string;
+  isTop?: boolean; // The widget at the top of the story has different styles
+};
+
+export const SocialSharing = ({
+  slug,
+  title,
+  isTop,
+  className,
+}: SocialSharingProps) => {
+  const [copiedText, copy] = useCopyToClipboard();
+  const [buttonState, setButtonState] = useState('default');
+
+  const handleCopyClick = () => {
+    copy(`${config.siteUrlProd}/${slug}`);
+    setButtonState('copied');
+
+    // Reset the button state after 5 seconds
+    setTimeout(() => {
+      setButtonState('default');
+    }, 5000);
+  };
```

Comment:
After 5 seconds, the copy link button resets itself to use the original icon.

Reply:
neat